public/app/plugins/datasource/jaeger/datasource.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  dateMath,
+  DateTime,
+  MutableDataFrame,
+  DataSourceApi,
+  DataSourceInstanceSettings,
+  DataQueryRequest,
+  DataQueryResponse,
+  DataQuery,
+} from '@grafana/data';
+import { BackendSrv, DatasourceRequestOptions } from 'app/core/services/backend_srv';
+import { TimeSrv } from 'app/features/dashboard/services/TimeSrv';
+
+import { Observable, from, of } from 'rxjs';
+
+function serializeParams(data: Record<string, any>) {
+  return Object.keys(data)
+    .map(k => `${encodeURIComponent(k)}=${encodeURIComponent(data[k])}`)
+    .join('&');
+}
```

Comment:
I can see this becoming grafana/data util

Reply:
Can't find but i'm pretty sure we already have such function :D

Reply:
There is actually a https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams for this. Will change.